pynxtools/dataconverter/convert.py

```diff
@@ -86,30 +96,56 @@ def convert(input_file: Tuple[str, ...],
         nxdl_path = os.path.join(definitions_path, "contributed_definitions", f"{nxdl}.nxdl.xml")
         if not os.path.exists(nxdl_path):
             nxdl_path = os.path.join(definitions_path, "applications", f"{nxdl}.nxdl.xml")
+        if not os.path.exists(nxdl_path):
+            nxdl_path = os.path.join(definitions_path, "base_classes", f"{nxdl}.nxdl.xml")
         if not os.path.exists(nxdl_path):
             raise FileNotFoundError(f"The nxdl file, {nxdl}, was not found.")
 
-    nxdl_root = ET.parse(nxdl_path).getroot()
+    return ET.parse(nxdl_path).getroot(), nxdl_path
 
-    if undocumented:
-        logger.setLevel(UNDOCUMENTED)
+
+def transfer_data_into_template(input_file,
```

Comment:
Are these extra functions to help reduce the size of the bigger convert function?

Reply:
This function is used in https://github.com/FAIRmat-NFDI/nomad-measurements/pull/13 in [this line](thttps://github.com/FAIRmat-NFDI/nomad-measurements/blob/b22454f1f9c578ff0b7f21ca43a33aa0a0ec66f4/src/nomad_measurements/xrd/schema.py#L20) to populate the template without file writing, I bealieve.

Reply:
Ah okay. That makes sense.